src/dombeck_lab_to_nwb/azcorra2023/interfaces/picoscope_recordinginterface.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+from pathlib import Path
+
+from natsort import natsorted
+from neuroconv.datainterfaces.ecephys.baserecordingextractorinterface import BaseRecordingExtractorInterface
+from neuroconv.utils import FolderPathType
+from spikeinterface import ConcatenateSegmentRecording
+
+from dombeck_lab_to_nwb.azcorra2023.extractors import PicoscopeRecordingExtractor
+
+
+class PicoscopeRecordingInterface(BaseRecordingExtractorInterface):
+    Extractor = ConcatenateSegmentRecording
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        verbose: bool = True,
+        es_key: str = "ElectricalSeriesSync",
+    ):
+        """
+        Load and prepare raw data and corresponding metadata from the Piscoscope (.mat files).
+
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            The folder containing the .mat files from the Picoscope.
+        verbose : bool, optional
+            Allows for verbose output, by default True.
+        es_key : str, optional
+            The key to use for the ElectricalSeries, by default "ElectricalSeriesSync".
+        """
+
+        self.folder_path = Path(folder_path)
+        session_name = self.folder_path.stem
+        mat_files = natsorted(self.folder_path.glob(f"{session_name}_*.mat"))
+        assert mat_files, f"The .mat files are missing from {folder_path}."
+
+        recording_list = [PicoscopeRecordingExtractor(file_path=str(file_path)) for file_path in mat_files]
+
+        super().__init__(recording_list=recording_list, verbose=verbose, es_key=es_key)
+
+        custom_channel_names = [
+            "chMov",
+            "chRed",
+            "chGreen",
+            "Light",
+            "ch470",
+            "Reward",
+            "Licking",
+            "AirPuff",
+        ]
```

Comment:
Before I'm making the changes, do you think @alessandratrapani I should follow the same design as you had and create an `ElectricalSeries` for each channel here?

Keep in mind for this format I have to concatenate recordings, which I'm doing this way only once, but I'm adding only one `ElectricalSeries` object.

Reply:
Bearing this is mind, I uploaded a prototype file [20200129-0001.nwb](https://s3.console.aws.amazon.com/s3/object/catalystneuro-processing?region=us-east-2&bucketType=general&prefix=dombeck-lab-to-nwb/20200129-0001.nwb)

Reply:
Actually, now that I see your PR, I think that we should organize this kind of signals differently. Synch signals should all go together in the same ElectricalSeries object, but here the only signal that can be considered a sync signal is  "ch470". 

The following are actual analog recordings:
- A = velocity (chMov)
- B = fiber 2 fluorescence (from here on called channel red - chR)
- C = fiber 1 fluorescence (channel green - chG)
- G = licking sensor output

And the rest are stimulus delivery signals:
- D = light stimulus trigger
- F = reward delivery trigger
- H = air puff delivery trigger

The easiest approach is to create an ElectricalSeries object for each channel, but then you lose the reference to the electrodes table, don't you? (I notice this in my stub [file](https://catalystneuro-processing.s3.us-east-2.amazonaws.com/higley-lab-to-nwb/11222019_grabAM05_spont_onlyTTLSignals.nwb))

There must be examples from past conversion projects that we can use as references.

Reply:
I agree, it definitely feels weird to have `Electrodes` and `ElectroGroup` for these signals. 

> D-F-H should go in intervals I guess

I found how G, D, F, H (licking, light, reward, air puff) binary signals are processed [here](https://github.com/DombeckLab/Azcorra2023/blob/2819bd5b7a6021243c44dfd45b5b25fd24ae8122/Fiber%20photometry%20data%20analysis/Data%20pre%20processing/selectSignals_paper.m#L109C12-L109C48) from lab. 
I can use this threshold to identify the onset and offset of events probably.

Reply:
Update from the team meeting; we are going to store the analog signals as `ElectricalSeries` for now and the binary signals as events using `ndx-events`.